PIL/GifImagePlugin.py

```diff
@@ -352,10 +352,21 @@ def _save(im, fp, filename, save_all=False):
 
         first_frame = None
         append_images = im.encoderinfo.get("append_images", [])
+        if "duration" in im.encoderinfo:
+            duration = im.encoderinfo["duration"]
+        else:
+            duration = None
+        frame_count = 0
         for imSequence in [im]+append_images:
             for im_frame in ImageSequence.Iterator(imSequence):
```

Comment:
The for loop `for im_frame in ImageSequence.Iterator(imSequence):` only excute one time if the current `Image` object in `append_images` only contains one frame. So it can not use `enumerate()` to get current page number.